### PR TITLE
Show 'updated TOS' and other notices

### PR DESCRIPTION
**Pull request by v18**
This is an in-progress PR for #770. Currently it:
- has a fake API call that returns some dummy data
- saves the data to wp_options
- displays a notice with the data

## Next Steps:
- [ ] decide on object structure for list notices received & saved
- [ ] display only the latest notice from TOS
- [ ] save the last retrieved notice id in wp_options
- [ ] let user dismiss notice
- [ ] updating the wp_option with new data
- [ ] log: when notice is received 
- [ ] log: when notice is dismissed

## About the notices

Here are things we care about for each notice.
- id
- title
- body
- button text
- should_log_response
- ^ type: info, success, etc
- ^ subject: TOS, other
- ^ pages it should be displayed on ('all' for TOS)
- ^ people it should be displayed to ('store owner' for TOS)

^ means that those items will not be added now, but may be added later as needed

What will not be built in this iteration
- logic for showing multiple types of notices (TOS at the same time as non-TOS, for example, and non-TOS with other non-TOS)
- notice types: info/success, and tos/non-tos
- pages to display notice on

The object received will probably look like:
```
{
  tos: [
    {
      id: 1,
      title: '',
      body: '',
      should_log_dismissal: true,
      can_display_with_others: false,
      type: 'info',
      pages: 'all',
      people: ['store-owner']
    }
  ],
  others: [...]
}
```

### CODE REVIEW

**Comment by v18** [comment]
Will have a look at the failing checks tomorrow :)

File: classes/class-wc-connect-remote-notices-store.php [hunk 1]
```diff
@@ -0,0 +1,41 @@
...
+		}
+
+		public function set_notices( $notices ) {
+			update_option( 'wc_connect_remote_notices', $notices );
+		}
+
+		public function fetch_notices_from_connect_server() {
+			$response = $this->api_client->get_notices( 1, 'en-us' );
+			// TODO: save more than one notice
+			$notices = (object) [
```
Comment (jeffstieler) [inline classes/class-wc-connect-remote-notices-store.php]: Use `array()` here instead of `[]` (since it's PHP 5.4.x and up).

File: uninstall.php [hunk 2]
```diff
@@ -0,0 +1,7 @@
+<?php
+// if uninstall.php is not called by WordPress, die
+if ( ! defined( 'WP_UNINSTALL_PLUGIN' ) ) {
+	die;
+}
+
+delete_option( 'wc_connect_remote_notices' );
```
Comment (jeffstieler) [inline uninstall.php]: Nice! This is an area we're currently deficient in. See: https://github.com/Automattic/woocommerce-connect-client/issues/776

File: classes/class-wc-connect-remote-notices-store.php (hunk 1 above)
Comment (v18) [inline classes/class-wc-connect-remote-notices-store.php]: Thanks! I also see this is the reason the automated tests failed for 5.3 and 5.2. Fixed, rebased, and pushed.

**Comment by v18** [comment]
Not working on this for the foreseeable future, so going to close the PR for now.

**Pull request closed without merging**